sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -144,6 +144,12 @@ class Analyzer(catalog: Catalog,
               failAnalysis(
                 s"unresolved operator ${operator.simpleString}")
 
+            case p @ Project(exprs, _) if exprs.length > 1 && exprs.flatMap(_.collect {
+              case e: Generator => true
+            }).length >= 1 =>
+              failAnalysis(
+                s"only a single generator allow in Projection ${operator.simpleString}")
```

Comment:
`Only a single table generating function is allowed in a SELECT clause, found: ${exprs.map(_.prettyString).mkString(",")}`